src/pretix/base/models/log.py

```diff
@@ -261,3 +202,134 @@ def bulk_postprocess(cls, objects):
         to_wh = [o.id for o in objects if o.webhook_type]
         if to_wh:
             notify_webhooks.apply_async(args=(to_wh,))
+
+
+class LogEntryTypeRegistry(EventPluginRegistry):
+    def new_from_dict(self, data):
+        def reg(clz):
+            for action_type, plain in data.items():
+                self.register(clz(action_type=action_type, plain=plain))
+        return reg
+
+
+log_entry_types = LogEntryTypeRegistry({'action_type': lambda o: getattr(o, 'action_type')})
+
+
+class LogEntryType:
+    def __init__(self, action_type=None, plain=None):
+        assert self.__module__ != LogEntryType.__module__  # must not instantiate base classes, only derived ones
```

Comment:
wouldn't `type(self) != LogEntryType` be more correct here?

Reply:
no, because I want all the "base types" (which are all types declared in this file) to be "abstract" / not directly instantiable. the main reason for that is that the __module__ is later used for checking in which plugin a LogEntryType is defined, so we can [mark the disabled ones separately](https://github.com/pretix/pretix/pull/4235/files/4e2799ff6570e2bb77920a1a54161293d642791b#diff-2bd781e18aa91b7ca1ab4364813c7938c72728986aec122af0ed13bc9f1a70feR62). I'll add that as a comment, because I had to think a bit to remember the reason myself :sweat_smile: